FormalConjectures/ForMathlib/Combinatorics/Basic.lean

```diff
@@ -89,32 +81,54 @@ theorem IsSidon.insert {A : Set α} {m : α} [IsRightCancelAdd α] [IsLeftCancel
     · simp_all
       exact fun _ _ _ _ _ ↦ by simp_all [add_comm]
 
+end Set
+
+namespace Finset
+
+@[simp, push_cast]
+theorem isSidon_toSet {A : Finset α} : IsSidon A.toSet ↔ IsSidon A := by
```

Comment:
I think this theorem should be also be proven for general objects that have the `SetLike` instance (and we can then remove the Finset special case once leanprover-community/mathlib4#28241 is merged)